File: Python/소프티어/GBC.py
import sys

n, m = list(map(int,sys.stdin.readline().split()))

limit = [0] * 101
p = 0
for _ in range(n):
    d, r = list(map(int,sys.stdin.readline().split()))
    for j in range(p, p+d):
        limit[j] = r
    p += d

p = 0
answer = 0
for _ in range(m):
    d, r = list(map(int,sys.stdin.readline().split()))
    for j in range(p, p+d):
        if limit[j] < r:
            answer = max(answer, r-limit[j])
    p += d

print(answer)

File: AlgorithmReview/softeer/GBC.swift
//
//  GBC.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2024/02/02.
//

import Foundation

let size = readLine()!.split(separator: " ").map{ Int($0)! }
let (n, m) = (size[0], size[1])

var limit = [Int](repeating: 0, count: 101)
var p = 1
for _ in 0..<n {
    let data = readLine()!.split(separator: " ").map{ Int($0)! }
    for j in p..<p+data[0] {
        limit[j] = data[1]
    }
    p = p+data[0]
}

p = 1
var answer = 0
for _ in 0..<m {
    let data = readLine()!.split(separator: " ").map{ Int($0)! }
    for j in p..<p+data[0]{
        if limit[j] < data[1] {
            answer = max(answer, data[1]-limit[j])
        }
    }
    p = p+data[0]
}

print(answer)

/*
Test 1
3 3
50 50
40 40
10 30
60 76
18 28
22 50
*/

/*
Test 2
3 3
50 90
10 90
40 50
50 40
10 100
40 40
*/
